go/common/entity/entity.go

```diff
@@ -77,6 +77,11 @@ func Load(baseDir string, signerFactory signature.SignerFactory) (*Entity, signa
 		return nil, nil, err
 	}
 
+	if ent.ID != signer.Public() {
```

Comment:
While this works as the current implementation is a fixed-size array, please use `!ent.ID.Equal(signer.Public())` to be consistent with other parts.